stdlib/websub/src/main/ballerina/websub/commons.bal

```diff
@@ -348,6 +348,58 @@ public type Notification object {
 
 };
 
+# Function to retrieve hub and topic URLs from the `http:response` from a publisher to a discovery request.
+#
+# + response - The `http:Response` received
+# + return - `(topic, hubs)` if parsing and extraction is successful, `error` if not
+public function extractTopicAndHubUrls(http:Response response) returns (string, string[])|error {
+    string[] linkHeaders;
+    error websubError;
+    if (response.hasHeader("Link")) {
+        linkHeaders = response.getHeaders("Link");
+    }
+
+    if (lengthof linkHeaders > 0) {
```

Comment:
The breath of the code segment can reduce by checking the else part first and return from there.
```
if (lengthof linkHeaders <= 0) {
   return { message: "Link header unavailable in discovery response" };
}
int hubIndex = 0;
string[] hubs;
....

```